src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx

```diff
@@ -102,129 +100,89 @@ const defaultProps = {
- *    - for this I only need the root jurisdiction
+ * 1).
  * 2). also gets the locations that are already assigned to plan
  *    - that's why I passed the plan as a prop
  * 3). render the drill down table.
  */
+/** 2 cases for this:
+ * 1). used as part of the autoSelection process.
+ *    when autoselection is enabled we finish the workflow in the plan assignment view,otherwise one would have to start all over.
+ *    for the time being there will not be auto-selecting existing jurisdiction assignments.
+ * 2). used when autoSelection is disabled.
+ *    here we have to show previous selections.
+ *    when autoselection is disabled autoSelect previously selected jurisdictions i.e. if plan is still draft.
+ */
+/** the business logic involved around the action handlers should remain in this component.
+ * This compoent will effectively be a HOC that specifies what the action handlers are
+ * and passes those on to the table. The table should be thought of as the presentational
+ * component.
+ */
 const JurisdictionTable = (props: JurisdictionSelectorTableProps) => {
   const {
     rootJurisdictionId,
-    treeFetchedCreator,
     currentParentNode,
     currentChildren,
     selectNodeCreator,
     deselectNodeCreator,
-    autoSelectNodesCreator,
     fetchPlanCreator,
     selectedLeafNodes,
-    leafNodes,
     plan,
     jurisdictionsMetadata,
     serviceClass,
+    autoSelectionFlow,
+    deselectAllNodesCreator,
   } = props;
-  /** helper function that decides which action creator to call when
-   * changing the selected status of a node
-   * @param nodId - id of the node of interest
-   * @param value - change selected status of node with said id to value
-   */
-  function applySelectedToNode(nodeId: string, value: boolean) {
-    if (value) {
-      selectNodeCreator(rootJurisdictionId, nodeId);
-    } else {
-      deselectNodeCreator(rootJurisdictionId, nodeId);
-    }
-  }
 
   /** function for determining whether the jurisdiction assignment table allows for
    * multiple selection or single selection based on the plan intervention type
+   *
+   * for planTypes that require single jurisdiction we will :
+   *  enabling a checkbox will be bound by these rules
+   *    1) if checkbox is a parent checkbox(i.e. if not belonging to a leaf node) disable.
    */
-  const isSingleSelect = () => {
-    const interventionType = plan.useContext.find(
-      element => element.code === INTERVENTION_TYPE_CODE
-    );
-    if (
-      interventionType &&
-      (interventionType.valueCodableConcept === InterventionType.FI ||
-        interventionType.valueCodableConcept === InterventionType.DynamicFI)
-    ) {
-      return true;
-    } else {
-      return false;
-    }
+  const isSingleSelect = (thePlan: PlanDefinition) => {
+    const interventionTypes = [
+      InterventionType.DynamicFI,
+      InterventionType.FI,
+      InterventionType.MDA,
+      InterventionType.MDAPoint,
+      InterventionType.DynamicMDA,
+    ];
```

Comment:
Can this not be extracted from `InterventionType` dynamically?  I worry that doing it this way is brittle i.e. adding new plan types suddenly means we need to remember to come and edit this code. I think it might be the case the enum types can work with Object.values or Object.keys

Reply:
:thinking:

Reply:
Moved this out to an env.

Reply:
> Moved this out to an env.

@p-netm any reason why we need an env as opposed to doing something like `Object.keys(InterventionType)`?

Reply:
we would still need to define somewhere, the plan types that should be excluded from the  resulting list, yes? in this case `IRS` and `Dynamic-IRS`? or maybe I still don't quite understand the requested change.